### PR TITLE
Data grid implementations for remaining reports

### PR DESCRIPTION
**Pull request by mail2bishnoi**
Hi Anton,

I have implemented Data-Grid for the remaining reports. Please review and suggest improvements.

For some of the reports, I have used 'getRowId' props for DataGrid because we have to provide a unique id for each row in the data set. 
https://github.com/mui-org/material-ui-x/issues/378

Thanks with regards.

### CODE REVIEW

File: modern/src/reports/TripReportPage.js [hunk 1]
```diff
@@ -30,40 +30,93 @@ const TripReportPage = () => {
...
-            ))}
-          </TableBody>
-        </Table>
-      </TableContainer>
+      <DataGrid
+        rows={items} 
+        columns={columns} 
+        hideFooter 
+        autoHeight
+        getRowId={() => Math.random()} />
```
Comment (tananaev) [inline modern/src/reports/TripReportPage.js:119]: Why not just use index for id?

File: modern/src/reports/EventReportPage.js [hunk 2]
```diff
@@ -59,30 +59,36 @@ const EventReportPage = () => {
...
-                <TableCell>{formatPosition(item, 'eventTime')}</TableCell>
-                <TableCell>{item.type}</TableCell>
-                <TableCell>{}</TableCell>
-                <TableCell>{}</TableCell>
-              </TableRow>
-            ))}
-          </TableBody>
-        </Table>
-      </TableContainer>
+      <DataGrid
```
Comment (tananaev) [inline modern/src/reports/EventReportPage.js:87]: Please add paper to the background. See route report. Same for other reports.

Comment (mail2bishnoi) [reply modern/src/reports/EventReportPage.js]: Done.

File: modern/src/reports/TripReportPage.js (hunk 1 above)
Comment (mail2bishnoi) [reply modern/src/reports/TripReportPage.js]: Sorry, I didn't understand which index. 

I think we can pass row object to getRowId callback but row doesn't have index.

Comment (tananaev) [reply modern/src/reports/TripReportPage.js]: We cannot get item index in the list?

Comment (mail2bishnoi) [reply modern/src/reports/TripReportPage.js]: Tried witout success.

File: modern/src/reports/EventReportPage.js [hunk 3]
```diff
@@ -59,30 +59,38 @@ const EventReportPage = () => {
...
-                <TableCell>{formatPosition(item, 'eventTime')}</TableCell>
-                <TableCell>{item.type}</TableCell>
-                <TableCell>{}</TableCell>
-                <TableCell>{}</TableCell>
-              </TableRow>
-            ))}
-          </TableBody>
-        </Table>
-      </TableContainer>
+      <Paper>
```
Comment (tananaev) [inline modern/src/reports/EventReportPage.js:87]: Can we move paper part to the `ReportLayoutPage` instead of duplicating it?

Comment (mail2bishnoi) [reply modern/src/reports/EventReportPage.js]: Yes. Done

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**